core/products/future.go

```diff
@@ -65,65 +65,59 @@ type oracle struct {
 	unsubscribe                     oracles.Unsubscriber
 	binding                         oracleBinding
 	data                            oracleData
-	settlementDataDecimals          uint32
 }
 
 type oracleData struct {
-	settlementData    *num.Uint
+	settlData         *num.Numeric
 	tradingTerminated bool
 }
 
-func (d *oracleData) SettlementData() (*num.Uint, error) {
-	if d.settlementData == nil {
+// SettlementData returns oracle data settlement data scaled as Uint.
+func (o *oracleData) SettlementData(op, ap uint32) (*num.Uint, error) {
+	if o.settlData.Decimal() == nil && o.settlData.Uint() == nil {
```

Comment:
Given that these functions should return a copy/clone, it's probably slightly more efficient to have `IsUint` and `IsDecimal` methods here, which just `return n.uint != nil` and `return n.dec != nil`, so we can check the wrapped numeric type without needing to clone the value

Reply:
This has been addresses, thanks